### PR TITLE
Improve homepage stickers

### PR DESCRIPTION
**Pull request by PonomareVlad**
To fix this:

[](https://github.com/grammyjs/website/blob/3e4a339c0ec777d15ead371b900e54d182882fa6/site/docs/public/icons/beach-animation.webp) [](https://github.com/grammyjs/website/blob/3e4a339c0ec777d15ead371b900e54d182882fa6/site/docs/public/icons/palette-animation.webp) [](https://github.com/grammyjs/website/blob/3e4a339c0ec777d15ead371b900e54d182882fa6/site/docs/public/icons/rocket-animation.webp)

### Result:
\- 193 KB from WEBP images
\+ 103 KB for Lottie Player (lazy loaded)
\= 90 KB saved and all glitches removed ✨

### CODE REVIEW

**Comment by CikiMomogi** [comment]
Have you tested it on a mobile device? The animation seems quite laggy on my end.

**Comment by PonomareVlad** [comment]
> Have you tested it on a mobile device? The animation seems quite laggy on my end.

Now testing, I'll be back a little later with the results

**Comment by roj1512** [comment]
It is fine with Safari on iOS.

**Comment by roj1512** [comment]
They are broken in translations.

**Comment by CikiMomogi** [comment]
The performance scores also drop a lot: 
- This PQ: 42 for mobile and 70 for desktop ([link to the test result](https://pagespeed.web.dev/analysis/https-652bbd2b34035d5bc2a4403a--grammy-netlify-app/6sbfo8px85?form_factor=mobile))
- Current: 91 for mobile and 99 for desktop ([link to the test result](https://pagespeed.web.dev/analysis/https-grammy-dev/go87ujc2n0?form_factor=desktop))

**Comment by PonomareVlad** [comment]
> They are broken in the translations.

Fixed

File: site/docs/.vitepress/config.ts [hunk 1]
```diff
@@ -82,4 +82,15 @@ export default defineConfig({
       chunkSizeWarningLimit: 1600,
     },
   },
+
```
Comment (roj1512) [inline site/docs/.vitepress/config.ts:85]:
```suggestion
```

**Comment by PonomareVlad** [comment]
> The performance scores also drop a lot:
> 
> * This PQ: 42 for mobile and 70 for desktop ([link to the test result](https://pagespeed.web.dev/analysis/https-652bbd2b34035d5bc2a4403a--grammy-netlify-app/6sbfo8px85?form_factor=mobile))
> * Current: 91 for mobile and 99 for desktop ([link to the test result](https://pagespeed.web.dev/analysis/https-grammy-dev/go87ujc2n0?form_factor=desktop))

With delayed import of Lottie player we get 93 for mobile and 97 for desktop ([link to test result](https://pagespeed.web.dev/analysis/https-652c01c6603f7a05db0d1be9--grammy-netlify-app/ij39ixwxch?form_factor=desktop))

**Review by roj1512 — COMMENTED** [review]
Animations are not playing in other languages.

**Comment by roj1512** [comment]
To be precise: when you change the language.

**Comment by PonomareVlad** [comment]
> Animations are not playing in other languages.

Yep, this case in line, currently I'm trying to avoid Lighthouse bias 🌚

**Comment by PonomareVlad** [comment]
> Animations are not playing in other languages.

Fixed

**Comment by PonomareVlad** [comment]
@CikiMomogi [this performance](https://pagespeed.web.dev/analysis/https-652c2ccb603f7a21ac0d1c26--grammy-netlify-app/5gjmm0c7o0?form_factor=mobile) is suitable ?

**Comment by PonomareVlad** [comment]
Also we can use Cache API, Network Information API and maybe Self Profiling API for conditionally disabling animations, but in next PQ 😏

**Comment by CikiMomogi** [comment]
I apologize if I misunderstood, but wouldn't adding an artificial timeout like this be counterproductive to our goal?

From my tests on desktop and Android, it seems that on desktop, the current website (grammy.dev) loads and runs the animated icon almost instantly, in contrast to this pull request. Unfortunately, the performance on Android isn't as smooth as mentioned earlier, the animation lags, likely due to my low-end device. However, during testing, grammy.dev didn't experience lag on the same device.

These test results are likely **not accurate**, so we need to test it on other low-end devices.

I apologize, but based on these tests, at the moment I'm hesitant to approve the use of a third-party player for .tgs files unless it doesn't impact performance, in which case, I'd gladly support your pull request. 👍

**Comment by PonomareVlad** [comment]
@CikiMomogi What about idea of using static SVGs for weak devices and load Lottie animations only for devices with high performance ?

**Comment by CikiMomogi** [comment]
Does this pull request feel faster on high-end devices compared to the current one?

**Comment by CikiMomogi** [comment]
I mean, wouldn't using WebM alone fix the issue, considering it's already supported on many devices?

**Comment by PonomareVlad** [comment]
> ... considering it's already supported on many browsers?

[SVG](https://caniuse.com/svg) is supported in more browsers than [WebP](https://caniuse.com/webp).

> I mean, wouldn't using WebP alone fix the issue ...

You can fix glitches in current WebP images ?

**Comment by CikiMomogi** [comment]
> [SVG](https://caniuse.com/svg) is supported in more browsers than [WebP](https://caniuse.com/webp).
But we cannot animate the icon in SVG (?)

> You can fix glitches in current WebP images ?
Which glitches?

**Comment by PonomareVlad** [comment]
> > [SVG](https://caniuse.com/svg) is supported in more browsers than [WebP](https://caniuse.com/webp).
> 
> But we cannot animate the icon in SVG (?)

We can but this will be more unstable (laggy) than using Lottie (which also uses SVG renderer with many optimisations)

**Comment by PonomareVlad** [comment]
> > You can fix glitches in current WebP images ?
> 
> Which glitches?

[This](https://t.me/grammyjs/175948)

**Comment by roj1512** [comment]
I will be reviewing after @CikiMomogi.

**Comment by CikiMomogi** [comment]
(https://github.com/grammyjs/website/pull/921/commits/3f7bcd9218c8af517fa75ff2b18f2ac05623d50c) https://652d5565c02bca26c8d763cc--grammy.netlify.app/ looks pretty good to me. Do you want to continue the workaround for low-end devices as we discussed in the Telegram group?

**Comment by PonomareVlad** [comment]
> Do you want to continue the workaround for low-end devices as we discussed in the Telegram group?

We can use something reliable like [this](https://github.com/pmndrs/detect-gpu).

I can implement some wrapper component over `<tgs-player>` with features like **preloader** and **performance target**.

For example:

```html
<awesome-tgs-player
    src="path/to/animation.tgs"
    preloader="path/to/image.svg"
    min-tier="1"
></awesome-tgs-player>

<!-- or -->

<awesome-tgs-player src="path/to/animation.tgs" min-tier="1">
    <img src="path/to/image.svg" slot="preloader">
</awesome-tgs-player>
```

But this will be require one more fetch to request GPU score for detected model and adds 24 KB to dependencies.

We can also consider a [more modern API](https://github.com/WICG/js-self-profiling), but I don't know how long it will take to implement scoring in this way.

**Comment by PonomareVlad** [comment]
[Prototype of wrapper component ready](https://github.com/PonomareVlad/lazy-lottie-player/blob/main/lazy-tgs-player.mjs) following [this variant](https://t.me/grammyjs/176785)

**Comment by CikiMomogi** [comment]
Can you build the preview website so I can test it on my devices?

**Comment by KnorpelSenf** [comment]
The preview is built at https://github.com/grammyjs/website/pull/921#issuecomment-1763345562

**Comment by CikiMomogi** [comment]
Oh, this one https://652f22b4fd970b15bedf5652--grammy.netlify.app?

**Comment by KnorpelSenf** [comment]
Yes

**Comment by CikiMomogi** [comment]
*sigh* Screw you GitHub Mobile!

**Comment by KnorpelSenf** [comment]
It only plays when you tap it but not as soon as the page loads (at least for me)

**Comment by PonomareVlad** [comment]
> It only plays when you tap it but not as soon as the page loads (at least for me)

Yes, forgot to take that into account, I'll fix it

**Comment by KnorpelSenf** [comment]
Wait what? But that's the desired behaviour, isn't it? We said we didn't want to see it playing automatically, so it's all good right now, no?

**Comment by KnorpelSenf** [comment]
The only thing is that it seems to play right after page load for @CikiMomogi but idk why

**Comment by CikiMomogi** [comment]
> The only thing is that it seems to play right after page load for @CikiMomogi but idk why

I didn't realize I was tapping on them while scrolling the page, which triggered the animation. :P

**Comment by PonomareVlad** [comment]
> Wait what? But that's the desired behaviour, isn't it? We said we didn't want to see it playing automatically, so it's all good right now, no?

I mean disabling this animations for devices with touch screen, because tap (mobile click) can freezes on some card after scrolling and animation will stay in play.

This behavior may be considered as visual bug by users and broke the first impression for someone.

**Comment by PonomareVlad** [comment]
But, if you want animations on mobile/touch devices we can consider any compatible trigger, like [scroll position](https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API).

**Comment by PonomareVlad** [comment]
> Also, I noticed that when you move the cursor to the outside the element before it plays (before the lottie player fully loaded), the animation will play forever until you hover over it again.

Yes, that's why I wrote a "prototype".

Perhaps this element will be useful to someone else (me), so I want to equip it with some options for customizing the behavior, but it will be later.

Now I will prepare the final version and fix the problems I noticed.

**Comment by KnorpelSenf** [comment]
Alright, we can disable it on mobile.

**Comment by PonomareVlad** [comment]
> Alright, we can disable it on mobile.

Done

**Review by CikiMomogi — APPROVED** [review]
LGTM! Thanks

**Review by roj1512 — APPROVED** [review]
Thanks!

**Pull request merged**